### DISCUSSION TITLE
How to change the math font?

### DISCUSSION THREAD

**Discussion by patrick-kidger** [typst/typst#998] [General]
I understand how to change the font for normal text, using a `#set text(font: ...)` directive. How can I do the same for mathematics?

The context is that I'm bumping up against #144, and need to change my font to one which makes the emojis appear. (Yes, I'm trying to use emojis in my mathematics, no don't question it... :D )

I can do this for normal text via `#set text(font: ("Foo-without-emojis", "Bar-with-emojis"))`, but I don't see a way to do it for mathematics.

**Comment by PgBiel** [comment]
For math there is `#show math.equation: set text(font: "Font here")`

Though, you may wish to use something like this instead:
```
#let emoji(content) = text(font: "Font-with-emojis", content)

$ x + emoji(👍) $
```

**Reply by patrick-kidger** [reply]
Great, thank you!
Do you know if there is a way to also get the current font? So that I can do `set text(font: (current-math-font, "emoji font"))`?

**Reply by PgBiel** [reply]
> Do you know if there is a way to also get the current font?

I think not, other than you manually storing in a variable every time you change the font (maybe using `state`). You could open an issue though so the possibility of adding such a feature can be investigated :eyes:

**Reply by PgBiel** [reply]
It is worth mentioning, however, that the default font (at least the general one) is `Linux Libertine`

**Reply by patrick-kidger** [reply]
Sounds good, thank you. I've opened #1003 to track this.
